src/updateIds/updateIds.js

```diff
@@ -13,82 +13,99 @@ const { parseTest, parseSuite, replaceSuiteTitle } = require('./helpers');
  * @param {*} opts
  * @returns
  */
-function updateIdsCommon(testData, testomatioMap, workDir, opts = {}) {
+function updateIdsCommon(testData, testomatioMap, workDir = {}) {
   const files = [];
   let duplicateTests = 0;
   let duplicateSuites = 0;
 
   debug('Test data:', testData);
 
   for (const testArr of testData) {
-    if (!testArr.length) continue;
+    if (!testArr || !testArr.length) continue;
 
     const file = `${workDir}/${testArr[0].file}`;
     debug('Updating file: ', file);
     let fileContent = fs.readFileSync(file, { encoding: 'utf8' });
+    const processedSuites = new Set();
 
-    const suite = testArr[0].suites[0] || '';
-
-    if (suite) {
-      const suiteIndex = suite;
-      const suiteWithoutTags = suite.replace(TAG_REGEX, '').trim();
-
-      const currentSuiteId = parseSuite(suiteIndex);
-      if (
-        currentSuiteId &&
-        testomatioMap.suites[suiteIndex] !== `@S${currentSuiteId}` &&
-        testomatioMap.suites[suiteWithoutTags] !== `@S${currentSuiteId}`
-      ) {
-        debug(`   Previous ID detected in suite '${suiteIndex}'`);
-        duplicateSuites++;
+    for (const data of testArr) {
+      if (!data || !data.suites || !Array.isArray(data.suites)) {
+        debug('Invalid data or suites:', data);
         continue;
       }
-
-      if (testomatioMap.suites[suiteIndex] && !suite.includes(testomatioMap.suites[suiteIndex])) {
-        fileContent = replaceSuiteTitle(suite, `${suite} ${testomatioMap.suites[suiteIndex]}`, fileContent);
-        fs.writeFileSync(file, fileContent);
-      } else if (testomatioMap.suites[suiteWithoutTags] && !suite.includes(testomatioMap.suites[suiteWithoutTags])) {
-        fileContent = replaceSuiteTitle(suite, `${suite} ${testomatioMap.suites[suiteWithoutTags]}`, fileContent);
-        fs.writeFileSync(file, fileContent);
+      for (const suite of data.suites) {
+        if (!suite) continue;
+        const suiteIndex = suite || '';
```

Comment:
don't assign empty string; if suiteIndex is empty - skip

and you have already checked if suite exists on the line above